File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/Headquarters.cs
namespace OctoSupply.Api.Models;

public sealed record Headquarters(
    int HeadquartersId,
    string Name,
    string? Description,
    string? Address,
    string? ContactPerson,
    string? Email,
    string? Phone);

public sealed class CreateHeadquartersRequest
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Address { get; set; }
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
}

public sealed class UpdateHeadquartersRequest
{
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? Address { get; set; }
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
}

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/headquarters.py
from __future__ import annotations

from pydantic import BaseModel


class Headquarters(BaseModel):
    headquartersId: int
    name: str
    description: str | None = None
    address: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None


class CreateHeadquartersRequest(BaseModel):
    name: str
    description: str | None = None
    address: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None


class UpdateHeadquartersRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    address: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None


class HeadquartersMetrics(BaseModel):
    score: int
    average: float
    display: str
